pallets/protos/src/lib.rs

```diff
@@ -914,10 +948,37 @@ pub mod pallet {
 					if params.desc { cats.iter().rev().map(|x| x.clone()).collect() } else { cats };
 
 				for category in cats {
-					if params.categories.len() != 0 && !params.categories.contains(&category) {
-						continue
+					if params.categories.len() != 0 { 
+						let param_category = &params.categories.clone().into_iter().next();
+						match param_category {
+							// If Category == Trait, check if the search is by name or ID
+							Some(Categories::Trait(param_shard_trait)) => {
+								if let Categories::Trait(stored_shard_trait) = &category { 
+								
+									// search by trait name (the ID must be all zeros)
+									if (param_shard_trait.id == [0u8; 16] && param_shard_trait.name == stored_shard_trait.name) || 
+											(param_shard_trait.name.is_empty() && param_shard_trait.id == stored_shard_trait.id)
+									{
+										// OK. Found the category matching name OR id. Do nothing here.
```

Comment:
return?

Reply:
Or we collect multiple? Not sure I like an empty block

Reply:
I can invert the statement into negative one (i.e. `if !(....)` ), and put a `continue` inside..this was more readable/intuitive for me to understand.

I can change it, if you prefer.

Reply:
> I can invert the statement into negative one (i.e. `if !(....)` ), and put a `continue` inside..this was more readable/intuitive for me to understand.
> 
> I can change it, if you prefer.

no I don't mind, makes sense